types/artifacts.d.ts

```diff
@@ -182,6 +184,26 @@ declare global {
         params: {name: string; value: string}[];
       }
 
+      export interface IFrameElement {
+        /** The `id` attribute of the iframe. */
+        id: string,
+        /** The `src` attribute of the iframe. */
+        src: string,
+        /** The iframe's ClientRect. @see https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect */
+        clientRect: {
+          top: number;
+          bottom: number;
+          left: number;
+          right: number;
+          width: number;
+          height: number;
+        },
+        /** If the iframe or an ancestor of the iframe is fixed in position. */
+        isPositionFixed: boolean,
+        /** The Frame of the iframe, undefined if cross-origin. @see https://chromedevtools.github.io/devtools-protocol/tot/Page#type-Frame */
+        frame?: Crdp.Page.Frame,
```

Comment:
> that aside, I support the idea of including this type here in the artifact. Seems useful for any audit work.

I'm not super wild about it, but I can live with it :)

But:
- the type should be defined inline, not referencing the protocol so that (theoretical?) users of our types won't need the protocol types as a dependency
- Since it's going into `PublicGathererArtifacts` we should trim it to just the properties we need (with any others which would be ridiculous to not include) since ideally we should only be adding to this type in the future, not trimming it, even on breaking versions